src/jdk.jfr/share/classes/jdk/jfr/internal/event/EventConfiguration.java

```diff
@@ -28,21 +28,37 @@
 import jdk.jfr.EventType;
 import jdk.jfr.SettingControl;
 import jdk.jfr.internal.EventControl;
+import jdk.jfr.internal.EventInstrumentation;
 import jdk.jfr.internal.JVM;
 import jdk.jfr.internal.PlatformEventType;
+import jdk.jfr.internal.settings.Throttler;
 
 public record EventConfiguration(
     PlatformEventType platformEventType,
     EventType eventType,
     EventControl eventControl,
     SettingControl[] settings,
+    Throttler throttler,
     long id) {
 
     // Accessed by generated code in event class
     public boolean shouldCommit(long duration) {
         return isEnabled() && duration >= platformEventType.getThresholdTicks();
     }
 
+    // Accessed by generated code in event class. Used by:
+    // static boolean shouldThrottleCommit(long duration, long timestamp)
+    public boolean shouldThrottleCommit(long duration, long timestamp) {
+        return isEnabled() && duration >= platformEventType.getThresholdTicks() && throttler.sample(timestamp);
+    }
+
+    // Caller must of Event::shouldThrottleCommit must check enablement.
+    // Accessed by generated code in event class. Used by:
+    // static boolean shouldThrottleCommit(long timestamp)
+    public boolean shouldThrottleCommit(long timestamp) {
+        return throttler.sample(timestamp);
```

Comment:
Can we assert on isEnabled?

Reply:
Between the time of the enabled() check and this method, the state may change, so it could lead to false positives in testing.